guiEditor/src/diagram/workbench.tsx

```diff
@@ -214,6 +226,22 @@ export class WorkbenchComponent extends React.Component<IWorkbenchComponentProps
         this.props.globalState.workbench = this;
     }
 
+    determineMouseSelection(selection: Nullable<Control>) {
+        if (selection && this._selectedGuiNodes.length <= 1) {
+            // if we're still on the same main selection, got down the tree.
+            if (selection === this._selectedGuiNodes[0] || selection === this._mainSelection) {
+                selection = this._getParentWithDepth(selection, --this._selectionDepth);

```

Comment:
`--this._selectionDepth` could the expression be placed on the line before ? 

This is a simple Neat preventing devs to think twice about does -- happens before or after :-)

Reply:
also it sounds like here selectionDepth will always end up -1 independently of the number of iteration in the getParentWithDepth unlinke the max one ?

Reply:
> 

Yes. Should should I move it inside the function?